openedx/core/djangoapps/persistenttask/migrations/0001_initial.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='FailedTask',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('task_name', models.CharField(max_length=255, db_index=True)),
```

Comment:
I considered this, but opted against it, for a few reasons:

* Text fields are stored outside the table, which would slow down querying.
* The longest path to a tasks.py file we have is 59 characters (openedx/core/djangoapps/content/course_structures/tasks.py), which means that we'd need to radically change our directory structure, or have a task name of almost 200 characters to actually hit this limit.
* If we do hit the limit, the tasks will still get persisted without failure.
* We could still differentiate between tasks using first and last ~125 characters of the task name.  Extra logic to do so could be added then, (not now because YAGNI: You Ain't Gonna Need It).
* [Indexing a TEXT column requires an explicit index size.](http://stackoverflow.com/questions/2889827/indexing-a-mysql-text-column#2889835)  This index is a prefix.  If we ever did hit our 255 character limit (or whatever size we choose), the problems would be worse, because the index would be on the prefix only, not the prefix and suffix, which would drop the uniquely identifying task name, and keep only the path, which could contain multiple tasks.  Then the lookup would require not only a table scan, but comparing the out-of-table text blobs.